javascript/testing/testing-2.md

```diff
@@ -63,3 +70,12 @@ There are two solutions to the 'tightly coupled code' problem.  The first, and b
 4. Jest includes some _really_ handy mocking functions.  Read about them in the [official docs](https://facebook.github.io/jest/docs/en/mock-functions.html#content)
 5. And finally, if you wish, you can add Jest to your webpack setup.  Read about that process [here](https://facebook.github.io/jest/docs/en/webpack.html#content).
 </div>
+
+### Knowledge Check 
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class='knowledge-check-link' href='#tightly-coupled-code'>What is tightly coupled code?</a>
+- <a class='knowledge-check-link' href='https://medium.com/@jamesjefferyuk/javascript-what-are-pure-functions-4d4d5392d49c'>What are the two requirements for a function to be pure?</a>
+- <a class='knowledge-check-link' href='https://medium.com/@jamesjefferyuk/javascript-what-are-pure-functions-4d4d5392d49c'>What is an observable side effect in regards to a function?</a>
```

Comment:
Can you rephrase this question? If I'm a learner I'd be confused if "observable side effect" was a technical term. I'd also be a bit confused on what you mean by "in regards to a function".

Reply:
Yes, no problem.

Reply:
I'm thinking like something along the lines of "What is a "side effect", and why is it important to identify them when testing a function?"

Reply:
Ok, got it. 👍